File: packages/flutter_photo_retriever/photo_retriever_ios/ios/Classes/photos/option/PhotoCacheType.swift
//
//  PhotoCacheType.swift
//  Runner
//
//  Created by kimrlyunah on 2/20/24.
//

import Foundation

enum PhotoCacheType {
    case memory, local
    
    static func fromValue(value: Int) -> PhotoCacheType? {
        switch value {
        case 0:
            return PhotoCacheType.memory
        case 1:
            return PhotoCacheType.local
        default:
            return nil
        }
    }
    func toValue() -> Int {
        switch self {
        case PhotoCacheType.memory:
            return 0
        case PhotoCacheType.local:
            return 1
        }
    }
}

File: packages/flutter_photo_retriever/photo_retriever_platform_interface/lib/src/option/photo_cache_type.dart
/// 이미지 가져오는 방법 결정
/// memory: 바이너리 값을 리턴
/// local: 로컬에 이미지 저장 후 저장된 경로를 리턴
enum PhotoCacheType {
  memory(0), local(1)
  ;
  final int value;
  const PhotoCacheType(this.value);
}
